dash-pipeline/bmv2/dash_headers.p4

```diff
@@ -93,7 +152,98 @@ header ipv6_t {
 
 const bit<16> IPV6_HDR_SIZE=320/8;
 
+
+enum bit<16> dash_flow_enabled_key_t {
+    ENI_ADDR = (1 << 0),
+    VNI = (1 << 1),
+    PROTOCOL = (1 << 2),
+    SRC_IP = (1 << 3),
+    DST_IP = (1 << 4),
+    SRC_PORT = (1 << 5),
+    DST_PORT = (1 << 6)
+}
+
+enum bit<16> dash_flow_entry_bulk_get_session_mode_t {
+    SAI_DASH_FLOW_ENTRY_BULK_GET_SESSION_MODE_GRPC = 0,
+    SAI_DASH_FLOW_ENTRY_BULK_GET_SESSION_MODE_VENDOR = 1,
+    SAI_DASH_FLOW_ENTRY_BULK_GET_SESSION_MODE_EVENT = 2,
+    SAI_DASH_FLOW_ENTRY_BULK_GET_SESSION_MODE_EVENT_WITHOUT_FLOW_STATE = 3
+}
+
+enum bit<16> dash_flow_entry_bulk_get_session_filter_key_t {
+    INVAILD = 0,
+    FLOW_TABLE_ID = 1,
+    ENI_ADDR = 2,
+    IP_PROTOCOL = 3,
+    SRC_IP_ADDR = 4,
+    DST_IP_ADDR = 5,
+    SRC_L4_PORT = 6,
+    DST_L4_PORT = 7,
+    KEY_VERSION = 8
+}
+
+enum bit<8> dash_flow_entry_bulk_get_session_op_key_t {
+    FILTER_OP_INVALID = 0,
+    FILTER_OP_EQUAL_TO = 1,
+    FILTER_OP_GREATER_THAN = 2,
+    FILTER_OP_GREATER_THAN_OR_EQUAL_TO = 3,
+    FILTER_OP_LESS_THAN = 4,
+    FILTER_OP_LESS_THAN_OR_EQUAL_TO = 5
+}
+
+struct flow_table_data_t {
+    bit<16> id;
+    bit<32> max_flow_count;
+    bit<16> flow_enabled_key;
+    bit<32> flow_ttl_in_milliseconds;
+}
+
+header flow_key_t {
+    EthernetAddress eni_mac;
+    bit<16> vnet_id;
+    IPv4ORv6Address src_ip;
+    IPv4ORv6Address dst_ip;
+    bit<16> src_port;
+    bit<16> dst_port;
+    bit<8> ip_proto;
+    bit<7> reserved;
+    bit<1> is_ip_v6;
+}
+const bit<16> FLOW_KEY_HDR_SIZE=flow_key_t.minSizeInBytes();
+
+header flow_data_t {
+    bit<7> reserved;
+    bit<1> is_unidirectional;
+    bit<32> version;
+    dash_direction_t direction;
+    bit<16> tunnel_id;
+    bit<32> routing_actions;
+    dash_meter_class_t meter_class;
+}
+const bit<16> FLOW_DATA_HDR_SIZE=flow_data_t.minSizeInBytes();
+
+// dash packet metadata
+header dash_packet_meta_t {
+    dash_packet_source_t packet_source;
+    dash_packet_type_t packet_type;
+    dash_packet_subtype_t packet_subtype;
+    bit<16>     length;
+}
+const bit<16> PACKET_META_HDR_SIZE=dash_packet_meta_t.minSizeInBytes();
+
+#define DASH_ETHTYPE 0x876d
+#define DPAPP_MAC 0x02fe23f0e413
```

Comment:
will the MAC be changed on other people's machine?

Reply:
It's a temporary hardcode. Mark FIXME comment to fix it later.